kedro/templates/project/hooks/utils.py

```diff
@@ -0,0 +1,150 @@
+from pathlib import Path
+import shutil
+import click
+
+current_dir = Path.cwd()
+
+lint_requirements = "black\nruff\n"
+lint_pyproject_requirements = """
+[tool.ruff]
+select = [
+    "F",  # Pyflakes
+    "E",  # Pycodestyle
+    "W",  # Pycodestyle
+    "UP",  # pyupgrade
+    "I",  # isort
+    "PL", # Pylint
+]
+ignore = ["E501"]  # Black take care off line-too-long
+"""
+
+test_requirements = "pytest-cov~=3.0\npytest-mock>=1.7.1, <2.0\npytest~=7.2"
+test_pyproject_requirements = """
+[tool.pytest.ini_options]
+addopts = \"\"\"
+--cov-report term-missing \\
+--cov src/{{ cookiecutter.python_package }} -ra
+\"\"\"
+
+[tool.coverage.report]
+fail_under = 0
+show_missing = true
+exclude_lines = ["pragma: no cover", "raise NotImplementedError"]
+"""
+
+docs_pyproject_requirements = """
+docs = [
+    "docutils<0.18.0",
+    "sphinx~=3.4.3",
+    "sphinx_rtd_theme==0.5.1",
+    "nbsphinx==0.8.1",
+    "nbstripout~=0.4",
+    "sphinx-autodoc-typehints==1.11.1",
+    "sphinx_copybutton==0.3.1",
+    "ipykernel>=5.3, <7.0",
+    "Jinja2<3.1.0",
+    "myst-parser~=0.17.2",
+]
+"""
+
+
+def parse_add_ons_input(add_ons_str):
+    """Parse the add-ons input string.
+
+    Args:
+        add_ons_str: Input string from prompts.yml.
+
+    Returns:
+        list: List of selected add-ons as strings.
+    """
+    if add_ons_str == "all":
+        return ["1", "2", "3", "4", "5"]
+    if add_ons_str == "none":
+        return []
+
+    # Split by comma
+    add_ons_choices = add_ons_str.split(",")
+    selected = []
+
+    for choice in add_ons_choices:
+        if "-" in choice:
+            start, end = choice.split("-")
+            if int(start) > int(end):
+                message = f"'{add_ons_str}' is an invalid range for Project Add-Ons."
+                click.secho(message, fg="red", err=True)
+                raise ValueError(message)
```

Comment:
The error here is very similar to the one above in `cli/starters.py` and I can see it gets triggered twice with a full stacktrace (see below). I don't think we should be showing the stacktrace at all and one message should be enough. 

It's so small because I couldn't otherwise screenshot it:

![Screenshot 2023-09-19 at 15 17 16](https://github.com/kedro-org/kedro/assets/49397448/cfbd7a9b-d7b2-4a35-9235-cd4270e04bb1)

Reply:
Added in [e3c7de5](https://github.com/kedro-org/kedro/pull/2987/commits/e3c7de548deb3ec2764f732d586a21ea97dda100) - though maybe a bit clunky? Any thoughts on the implementation?

Reply:
Addressed in [472a57c](https://github.com/kedro-org/kedro/pull/2987/commits/472a57c85a2a987692e7506513305a10c628989d) - I tried isolating the ValueError from the stack trace to no avail (both `with_traceback(None)` and `sys.tracebacklimit = 0` were unsuccessful, as well as several other approaches), I settled on just using sys.exit(1). Let me know your thoughts on this approach.